lib/stages/ExampleDotProduct.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef EXAMPLE_DOT_PRODUCT_STAGE_H
+#define EXAMPLE_DOT_PRODUCT_STAGE_H
+
+#include "Stage.hpp"
+#include "buffer.h"
+#include "errors.h"
+#include "util.h"
+
+/**
+ * @class ExampleDotProduct
+ * @brief A stage to compute the dot product between two vectors: A and B, which are represented by
+ * two buffers. The result is written to an output buffer.
+ *
+ * @par Buffers
+ * @buffer in_a_buf The input buffer representing vector A.
+ *      @buffer_format standard
```

Comment:
```suggestion
 *      @buffer_format Array of floats
```
And same for the other buffers below.